app/src/main/java/com/example/wearableapp/presentation/ui/MeasureHeartRateActivity.kt

```diff
@@ -0,0 +1,149 @@
+import android.graphics.drawable.AnimationDrawable
+import android.hardware.Sensor
+import android.hardware.SensorEvent
+import android.hardware.SensorEventListener
+import android.hardware.SensorManager
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.lifecycleScope
+import com.example.domain.model.HeartRateData
+import com.example.wearableapp.R
+import com.example.wearableapp.databinding.ActivityMeasHeartRateBinding
+import com.example.wearableapp.presentation.adapter.HeartDataAdapter
+import com.example.wearableapp.presentation.service.HeartRateSensorService
+import com.example.wearableapp.presentation.utils.Constants.Companion.DEFAULT_NAME
+import com.example.wearableapp.presentation.viewmodel.MeasureHeartRateViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import kotlin.collections.ArrayList
+
+
+class MeasureHeartRateActivity : AppCompatActivity(), SensorEventListener, View.OnClickListener {
+
+    val TAG = MeasureHeartRateActivity::class.simpleName
+    private lateinit var binding: ActivityMeasHeartRateBinding
+
+    private val viewModel by viewModel<MeasureHeartRateViewModel>()
+
+    private var mSensorManager: SensorManager? = null
+    private var mHeartSensor: Sensor? = null
+
+    private var measureHRList = ArrayList<HeartRateData>()
+
+    var animator = AnimationDrawable()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        binding = ActivityMeasHeartRateBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        setListeners()
+        checkSensorIsAvailable()
+    }
+
+    private fun setListeners() {
+        binding.tvHeartRateStartId.setOnClickListener(this)
+        binding.tvHeartRateStopId.setOnClickListener(this)
+    }
+
+    private fun startAnimation(){
+        binding.floatLikeId.setBackgroundResource(R.drawable.animation_list)
+        animator = binding.floatLikeId.background as AnimationDrawable
+        animator.start()
+    }
+
+    private fun stopAnimation(){
+        animator.stop()
+        binding.floatLikeId.setBackgroundResource(R.drawable.ic_baseline_favorite_24)
+        binding.heratRateCount.text = "--"
+    }
+
+    private fun startMeasureHRView(){
+        binding.clHeartStartId.visibility = View.GONE
+        binding.clHeartStopId.visibility = View.VISIBLE
+        startAnimation()
+        mSensorManager?.registerListener(
+            this,
+            mHeartSensor,
+            SensorManager.SENSOR_DELAY_FASTEST
+        );
+    }
+
+    private fun stopMeasureHRView(){
+        stopAnimation()
+        binding.clHeartStartId.visibility = View.VISIBLE
+        binding.clHeartStopId.visibility = View.GONE
+        mSensorManager?.unregisterListener(this)
+    }
+
+    private fun getSensorData() {
```

Comment:
please move all sensor operation into service, we should be able to listen for events in ambient mode too

Reply:
working on it